server/src/test/java/org/elasticsearch/gateway/Zen2GatewayMetaStateTests.java

```diff
@@ -0,0 +1,235 @@
+        final IndexMetaData indexMetaData = createIndexMetaData(indexName, numberOfShards, version);
+        final ClusterState state = createClusterState(randomNonNegativeLong(),
+                MetaData.builder().coordinationMetaData(createCoordinationMetaData(term)).put(indexMetaData, false).build());
+        gateway.setLastAcceptedState(state);
+
+        gateway = maybeNew(gateway);
+        final long newTerm = randomValueOtherThan(term, () -> randomNonNegativeLong());
+        final int newNumberOfShards = randomValueOtherThan(numberOfShards, () -> randomIntBetween(1,5));
+        final IndexMetaData newIndexMetaData = createIndexMetaData(indexName, newNumberOfShards, version);
+        final ClusterState newClusterState = createClusterState(randomNonNegativeLong(),
+                MetaData.builder().coordinationMetaData(createCoordinationMetaData(newTerm)).put(newIndexMetaData, false).build());
+        gateway.setLastAcceptedState(newClusterState);
+
+        gateway = maybeNew(gateway);
+        assertThat(gateway.getLastAcceptedState().metaData().index(indexName), equalTo(newIndexMetaData));
+    }
+
+    public void testCurrentTermAndTermAreDifferent() throws IOException {
+        Gateway gateway = newGateway();
+
+        long currentTerm = randomNonNegativeLong();
+        long term  = randomValueOtherThan(currentTerm, () -> randomNonNegativeLong());
+
+        gateway.setCurrentTerm(currentTerm);
+        gateway.setLastAcceptedState(createClusterState(randomNonNegativeLong(),
+                MetaData.builder().coordinationMetaData(CoordinationMetaData.builder().term(term).build()).build()));
+
+        gateway = maybeNew(gateway);
+        assertThat(gateway.getCurrentTerm(), equalTo(currentTerm));
+        assertThat(gateway.getLastAcceptedState().coordinationMetaData().term(), equalTo(term));
+    }
+
+    public void testMarkAcceptedConfigAsCommitted() throws IOException {
+        Gateway gateway = newGateway();
+
+        CoordinationMetaData coordinationMetaData = createCoordinationMetaData(randomNonNegativeLong());
+        ClusterState state = createClusterState(randomNonNegativeLong(),
+                MetaData.builder().coordinationMetaData(coordinationMetaData).build());
+        gateway.setLastAcceptedState(state);
+
+        gateway = maybeNew(gateway);
+        assertThat(gateway.getLastAcceptedState().getLastAcceptedConfiguration(),
+                not(equalTo(gateway.getLastAcceptedState().getLastCommittedConfiguration())));
+        gateway.markLastAcceptedConfigAsCommitted();
+
+        gateway = maybeNew(gateway);
+        assertThat(gateway.getLastAcceptedState().getLastAcceptedConfiguration(),
+                equalTo(gateway.getLastAcceptedState().getLastCommittedConfiguration()));
+        gateway.markLastAcceptedConfigAsCommitted();
+
+        gateway = maybeNew(gateway);
+        assertThat(gateway.getLastAcceptedState().getLastAcceptedConfiguration(),
```

Comment:
also assert that nothing else has changed?

Reply:
https://github.com/elastic/elasticsearch/pull/35819/commits/971aef183925c61ee6b147ae5102ad951e3a52b8